src/workerd/io/compatibility-date.capnp

```diff
@@ -832,4 +832,15 @@ struct CompatibilityFlags @0x8f8c1b68151b6cef {
   # throw an error when unknown import attributes are encountered. In the new module registry
   # implementation the recommended behavior is what is implemented. With this compat flag
   # enabled, the original module registry implementation will follow the recommended behavior.
+
+  unsupportedProcessRealPlatform @95 :Bool
+      $compatEnableFlag("unsupported_process_real_platform")
+      $experimental;
+  # Enables support for process.platform to expose the actual system platform
+  # This will never ever be supported and is a WPT test path only.
```

Comment:
```suggestion
  # By default, Workerd will always expose "linux" as the process.platform
  # This flag ensures the support to enable internal use-cases that validates
  # the underlying operating system.
```

Reply:
Added a variant of this suggestion retaining the note that this will never be formally supported.